test/jdk/java/util/Locale/LocaleProviders.java

```diff
@@ -398,22 +398,31 @@ static void bug8232860Test() {
     }
 
     static void bug8245241Test(String expected) {
-        LogRecord[] lra = new LogRecord[1];
-        StreamHandler handler = new StreamHandler() {
+        // this will ensure LocaleProviderAdapter initialization
+        DateFormat.getDateInstance();
+        LogConfig.handler.flush();
+
+        var r = LogConfig.lra[0];
+        if (r == null ||
+            r.getLevel() != Level.INFO ||
+            !r.getMessage().equals(expected)) {
+            throw new RuntimeException("Expected log was not emitted. LogRecord: " +
+                    (r == null ? null : r.getMessage()));
+        }
+    }
+
+    // Set the root logger on loading the logging class
+    public static class LogConfig {
+        final static LogRecord[] lra = new LogRecord[1];
```

Comment:
I would suggest to use some multi-thread safe container rather than a simple array to store the LogRecord. For instance - a CopyOnWriteArrayList - or something like that. Also you may want to harden the test by allowing for the possibility that some other logging might have occurred, and search the list for the record you expect rather than assuming it will be the first and only one.

Reply:
Thanks, Daniel. Will update the fix as you suggested.